tests/test_barely_json.py

```diff
@@ -257,3 +259,5 @@ def test_custom_resolver(self):
             resolver=resolver
         )
 
+    def test_str_illegal(self):
```

Comment:
Great that you've added a test case! However, to keep with the existing style, all tests for `IllegalValue` should go into a separate `TestIllegalValue` class.